src/pages/index.js

```diff
@@ -1,90 +1,233 @@
+import PopupWithConfirmation from "../components/PopupWithConfirmation.js";
 import UserInfo from "../components/UserInfo.js";
-import {
-  initialCards,
-  validationSettings,
-  profileSelectors,
-} from "../utils/utils.js";
-import Card from "../components/Card.js";
-import FormValidator from "../components/FormValidator.js";
-import "../pages/index.css";
+import { initialCards, validationSettings } from "../utils/utils.js";
+import Api from "../components/Api.js";
 
-// Function to handle image click and open popup
-const handleImageClick = (name, link) => {
-  imagePopup.open({ name, link });
-};
+// DOM Elements
+const profileEditModal = document.querySelector("#profileEditModal");
+const addCardModal = document.querySelector("#add-card-modal");
+const addCardFormElement = addCardModal.querySelector(".modal__form");
+const avatarFormElement = document.querySelector("#change-profile-form");
+const addNewCardButton = document.querySelector(".profile__add-button");
+const profileEditButton = document.querySelector("#profile-edit-button");
+const profileTitleInput = document.querySelector("#profile-title-input");
+const profileDescriptionInput = document.querySelector(
+  "#profile-description-input"
+);
+const profileImageEditButton = document.querySelector(
+  ".profile-image__edit-button"
+);
 
-// Function to create a card
+// API Initialization
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "740df34c-ea9e-4ff2-ad05-85becca779ab",
+    "Content-Type": "application/json",
+  },
+});
+
+let cardList;
+let userInfo;
+
+// Load Initial Data
+api
+  .loadPageResults()
+  .then(([card, userData]) => {
+    // Initialize Card List
+    cardList = new Section(
+      {
+        items: [...initialCards, ...card],
+        renderer: createCard,
+      },
+      ".gallery__cards"
+    );
+    cardList.renderItems();
+
+    // Initialize User Info
+    userInfo = new UserInfo(
+      ".profile__title",
+      ".profile__description",
+      ".profile__image"
+    );
+    userInfo.setUserInfo({ title: userData.name, description: userData.about });
+    userInfo.setUserImage(userData.avatar);
+  })
+  .catch(console.error);
+
+// Function to Create Card
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  return card.getView();
+  const cardElement = new Card(
+    cardData,
+    "#card-template",
+    handlePreviewPicture,
+    handleDeleteClick,
+    handleLikeClick
+  );
+  return cardElement.getView();
 }
 
-// Initialize Cards Section
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      const cardElement = createCard(cardData);
-      cardSection.addItem(cardElement);
-    },
-  },
-  ".gallery__cards"
-);
-
-cardSection.renderItems();
+// Profile Edit Submit Handler
+function handleProfileEditSubmit(userData) {
+  profilePopupForm.setLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```